### PR TITLE
Make codegen helper functions into true module

### PR DESCRIPTION
**Pull request by texasaggie97**
[X] This contribution adheres to [CONTRIBUTING.md]
### What does this Pull Request accomplish?
* Move contents of helper.py and metadata_utilities.py into helper module
  * Module is now unit tested
  * This will lower our overall coverage because we are adding more code that is tested without complete coverage
* Partial fix for #108 
  * Allows for adding additional tests

### Why should this Pull Request be merged?
* Puts testing in place to improve coverage

### What testing has been done?
* Travis & System tests

### CODE REVIEW

File: src/nidmm/metadata/__init__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from metadata.attributes import attributes
 from metadata.enums import enums
 
-import build.metadata_utilities
+import build.helper as helper
```
Comment (marcoskirsch) [inline src/nidmm/metadata/__init__.py]: "build" and "helper" are very generic names.
I think it's time we name our code generator and have the module use that name. For example: say we need to break up the repo. You're in nidmm-python repo... what would you need to import?

What do you think of nimigen or nimi-gen ?

File: build/helper/__init__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from build.helper.codegen_helper import get_ctype_variable_declaration_snippet  # noqa: F401
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/__init__.py]: This provides two benefits:
1. Encapsulation - we only bring in the "public" functions
1. The functions we do bring in are in the helper package namespace. The __init__.py file can be changed to this:
    ``` python
    import build.helper.codegen_helper
    import build.helper.documentation_helper
    import build.helper.helper
    import build.helper.metadata_add_all
    import build.helper.metadata_filters
    import build.helper.metadata_find
    import build.helper.metadata_merge_dicts
    ```
    but then each call needs to add the module name:
    ``` python
    import build.helper as helper
    helper.metadata_merge_dicts.merge_dicts(...)
    ```

File: build/helper/helper.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    if sys.version_info.major < 3:
+        with add_to_path('build/templates'):
+            p_types = importlib.import_module('python_types')
+    else:
+        p_types = importlib.import_module('build.templates.python_types')
+    v_type = getattr(p_types, visa_type)
+
+    return type(v_type()).__name__
+
+
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/helper.py]: I am open to new names...

File: src/nidmm/metadata/__init__.py [hunk 4]
```diff
@@ -12,13 +12,13 @@
 from metadata.functions_addon import functions_buffer_info
 from metadata.functions_addon import functions_is_error_handling
 
-build.metadata_utilities.merge_dicts(functions, functions_codegen_method)
-build.metadata_utilities.merge_dicts(functions, functions_enums)
-build.metadata_utilities.merge_dicts(functions, functions_params_types)
-build.metadata_utilities.merge_dicts(functions, functions_buffer_info)
-build.metadata_utilities.merge_dicts(functions, functions_is_error_handling)
+helper.merge_dicts(functions, functions_codegen_method)
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply src/nidmm/metadata/__init__.py]: I have some ideas, but they would depend on the "addon" dictionaries follow a naming convention.

File: src/nidmm/metadata/__init__.py (hunk 1 above)
Comment (texasaggie97) [reply src/nidmm/metadata/__init__.py]: How about calling it marc? That way it's almost each of our names, but not quite...

File: build/helper/metadata_filters.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Filters
+def extract_codegen_functions(functions):
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_filters.py]: Done

File: build/helper/helper.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# TODO(marcoskirsch): This file should definitely not live here but I had trouble getting import to work.
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/helper.py]: Removed

File: build/helper/helper.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+# TODO(marcoskirsch): Figure out unit test for this.
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/helper.py]: Removed

File: build/helper/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcoskirsch) [reply build/helper/__init__.py]: ok

File: build/helper/helper.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcoskirsch) [reply build/helper/helper.py]: As per offline discussion, I think we need to take a step back and really figure out what this all looks like (file names, structures, templates inside build module? GNU makefiles inside build module?).
I'll open an issue.

**Pull request merged**